lib/src/convert.rs

```diff
@@ -286,6 +309,15 @@ impl<A: Into<BoltType> + Clone> From<&[A]> for BoltType {
     }
 }
 
+impl<A: Into<BoltString> + Clone, B: Into<BoltType> + Clone> From<HashMap<A, B>> for BoltType {
+    fn from(value: HashMap<A, B>) -> Self {
+        BoltType::Map(BoltMap { value: value.iter().fold(HashMap::new(), |mut accum, (k, v)| {
+            accum.insert(k.clone().into(), v.clone().into());
+            accum
+        })})
+    }
+}
+
```

Comment:
Three things I would like to change here:
* Since we convert from an owned value of HashMap, we can use `into_iter()` instead of `iter()` to avoid having to clone the value
* We can make use of the `FromIterator` implementation that exists for BoltMap
* Conventionally, generic type names for maps are usually `K` and `V`

```suggestion
impl<K: Into<BoltString>, V: Into<BoltType>> From<HashMap<K, V>> for BoltType {
    fn from(value: HashMap<K, V>) -> Self {
        BoltType::Map(value.into_ter().map(|(k, v)| (BoltString::from(k), BoltType::from(v))).collect())
    }
}

```

(`into` might work over `from` and maybe the parenthesis are off, I'm typing this in the browser right now, not an editor 😅)